p2p/discover/table.go

```diff
@@ -494,10 +500,23 @@ func (tab *Table) bucketAtDistance(d int) *bucket {
 //
 // The caller must not hold tab.mutex.
 func (tab *Table) addSeenNode(n *node) {
-	go tab.addSeenNodeSync(n)
+	select {
+	case <-tab.workerPoolTask:
+		go tab.addSeenNodeSync(n)
```

Comment:
I'm not so sure about this but when shutting down the node, table.close() is called and it seems like these goroutines are created does not sync with the shutdown event. I'm not sure if it safe, as I see some other old code does not sync with shutdown event either.

Reply:
I don't think we need to close the channel here, as documentation, the channel is cleaned up by GC, and force closing the channel in the close method maybe needed in some cases to terminate a loop range. For our cases, we don't have a loop here.

Reply:
I mean the concern is whether the code in goroutine can be safely executed when the table is closed